be/src/storage/lake/lake_primary_index.cpp

```diff
@@ -124,42 +130,51 @@ Status LakePrimaryIndex::_do_lake_load(Tablet* tablet, const TabletMetadata& met
     }
     // NOTICE: primary index will be builded by segment files in metadata, and delvecs.
     // The delvecs we need are stored in delvec file by base_version and current MetaFileBuilder's cache.
-    for (auto& rowset : *rowsets) {
-        auto res = rowset->get_each_segment_iterator_with_delvec(pkey_schema, base_version, builder, &stats);
-        if (!res.ok()) {
-            return res.status();
+    if (config::use_lake_pk_index_loader) {
+        auto future = ExecEnv::GetInstance()->lake_pk_index_loader()->load(tablet, *rowsets, pkey_schema, base_version,
+                                                                           builder, this);
+        auto st = future.get();
```

Comment:
Instead of submitting the tasks to a different thread pool and letting the publish version thread pool block and wait, why not just execute the load tasks in parallel in the publish version's thread pool?

Reply:
Used `connector_scan_executor`.